apps/enc.c

```diff
@@ -543,26 +570,92 @@ int enc_main(int argc, char **argv)
         }
     }
 
-    /* Only encrypt/decrypt as we write the file */
-    if (benc != NULL)
-        wbio = BIO_push(benc, wbio);
-
+    /* Read input to buffer BIO, because some ciphers can't be streamed into (usually AEAD) */
```

Comment:
I think it should only buffer AEAD ciphers to memory.  As enc stands it can deal with any sized file, with this change it is limited to less than memory sized.  Small embedded devices and multi terabyte files should both work.